Gems/Multiplayer/Code/Source/MultiplayerSystemComponent.cpp

```diff
@@ -855,6 +884,104 @@ namespace Multiplayer
         return true;
     }
 
+    bool MultiplayerSystemComponent::HandleRequest(
+        IConnection* connection,
+        [[maybe_unused]] const IPacketHeader& packetHeader,
+        MultiplayerPackets::SyncComponentMismatch& packet)
+    {
+        // We've received all the component data from this connection
+        // Iterate over each component and see what's been added, missing, or modified
+        for (const auto& [theirComponentName, theirComponentHash] : packet.GetComponentVersions())
+        {
+            // Check for modified components
+            AZ::HashValue64 localComponentHash;
+            if (GetMultiplayerComponentRegistry()->FindComponentVersionHashByName(theirComponentName, localComponentHash))
+            {
+                
+                if (theirComponentHash != localComponentHash)
+                {
+                    AZLOG_ERROR(
+                        "Multiplayer component mismatch! %s has a different version hash. Please make sure both client and server have "
+                        "matching multiplayer components.",
+                        theirComponentName.GetCStr());
+                }
+            }
+            else
+            {
+                // Connected machine is using a multiplayer component we don't have
+                AZLOG_ERROR(
+                    "Multiplayer component mismatch! We're missing a component with version hash 0x%llx. "
```

Comment:
nit: avoid personal pronouns in error messages. Unclear to reader who "we" is.

Reply:
Fixed!